src/main/webapp/src/app/admin-page/admin-page.component.ts

```diff
@@ -0,0 +1,341 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, OnInit} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {
+  HttpClient,
+  HttpResponse,
+  HttpErrorResponse,
+} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {Garden} from '../model/garden.model';
+import {Plant} from '../model/plant.model';
+import {User} from '../model/user.model';
+
+@Component({
+  selector: 'admin-page',
+  templateUrl: './admin-page.component.html',
+  styleUrls: [
+    './admin-page.component.css',
+    '../common/growpod-page-styles.css',
+  ],
+})
+
+/**
+ * This is a component designed to allow administrators to manage the gardens they
+ * own.
+ *
+ * This page takes an argument 'garden-id'. This is not optional, and determines
+ * the garden that is being administered. Only the garden owner can make changes to
+ * their garden.
+ */
+export class AdminPageComponent implements OnInit {
+  isLoaded = false;
+  gardenProfile: Garden | null;
+  gardenManager: string;
+
+  // User list
+  gardenUserIdList: Array<string> | null = null;
+  gardenUserNameMap: Map<string, string>;
+  gardenUserIdListError = 'User list not loaded';
+
+  // Plant list
+  gardenPlantIdList: Array<string> | null = null;
+  gardenPlantNameMap: Map<string, string>;
+  gardenPlantIdListError = '';
+  errorMessage = '';
+
+  /**
+   * Initializes the component based on provided arguments
+   *
+   * @param route Contains arguments.
+   */
+  constructor(private route: ActivatedRoute, private httpClient: HttpClient) {}
+
+  ngOnInit(): void {
+    const gardenIdArg = this.route.snapshot.paramMap.get('garden-id');
+    if (!gardenIdArg) {
+      this.gardenProfile = null;
```

Comment:
What is the value of `gardenProfile` when you set it to `null` here?

Reply:
In this situation, I can get away with just initializing `gardenProfile` where it's declared with `null`.